testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_database.py

```diff
@@ -362,3 +368,404 @@ def test_copy_prebuilt_database(params_from_base_test_setup, encrypted):
+
+    # 1b. prepare SGW, create 2 users on 2 different channels and set up connections
+    username1 = "autotest"
+    username2 = "autotest2"
+    password = "password"
+
+    sg_client = MobileRestClient()
+    sg_client.create_user(sg_admin_url, sg_db, username1, password=password, channels=channel1)
+    cookie1, session_id1 = sg_client.create_session(sg_admin_url, sg_db, username1)
+
+    sg_client.create_user(sg_admin_url, sg_db, username2, password=password, channels=channel2)
+    cookie2, session_id2 = sg_client.create_session(sg_admin_url, sg_db, username2)
+
+    # 2. start 2 replicators on 2 different channels, set all continous to True
+    replicator = Replication(base_url)
+    authenticator = Authenticator(base_url)
+
+    replicator_authenticator1 = authenticator.authentication(session_id1, cookie1, authentication_type="session")
+    repl_config1 = replicator.configure(source_db=cbl_db, replicator_authenticator=replicator_authenticator1, target_url=sg_blip_url,
+                                        replication_type="push_pull", continuous=True, channels=channel1)
+    replicator_authenticator2 = authenticator.authentication(session_id2, cookie2, authentication_type="session")
+    repl_config2 = replicator.configure(source_db=cbl_db, replicator_authenticator=replicator_authenticator2, target_url=sg_blip_url,
+                                        replication_type="push_pull", continuous=True, channels=channel2)
+
+    # start all replicators
+    repl1 = replicator.create(repl_config1)
+    replicator.start(repl1)
+
+    repl2 = replicator.create(repl_config2)
+    replicator.start(repl2)
+
+    # 3. register a live query to the cbl db
+    qy = Query(base_url)
+    query = qy.query_select_all(cbl_db)
+    query_listener = qy.addChangeListener(query)
+
+    log_info(replicator.getActivitylevel(repl1))
+    log_info(replicator.getActivitylevel(repl2))
+
+    # TODO: need to clarify with dev about close db on replicator state
+    # after clarification, this code suppose to be removed
+    replicator.wait_until_replicator_idle(repl1)
+    replicator.wait_until_replicator_idle(repl2)
+
+    log_info(replicator.getActivitylevel(repl1))
+    log_info(replicator.getActivitylevel(repl2))
+
+    # 4. close the database
+    # 5. verify the database is closed successfully
+    try:
+        log_info("closing database")
+        db.close(cbl_db)
+        log_info("database is closed successfully")
+        assert True
+    except KeyError:
+        qy.removeChangeListener(query_listener)
+        assert False, "closing database with active replicators are failed"
+
+
+@pytest.mark.listener
+@pytest.mark.hydrogen
+def test_db_delete_on_active_replicators(params_from_base_test_setup):
+    """
+        @summary:
+        1. create a db on cbl, have a sgw available
+        2. start 3 replicators with the sgw, ensure one of
+           the replicator is push_pull replicator with continues=true
+        3. delete the cbl db
```

Comment:
this scenario looks covered in our regression already .. Having replicators , then closed db after replication is done. is this test doing anything different from current regression? I want to avoid duplicate tests if there is any

Reply:
I removed wait_until_replicator_idle call, and close/delete database immediately. the wait_until_replicator_idle was here because these is a hanging issue to close database while replicator is at connecting state, CBL-988 is the ticket for the issue